src/components/SignupScreen.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import {addUser} from '../store/UsersActions';
+
+import {
+  Button,
+  StyleSheet,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+
+class SignupScreen extends Component {
+  constructor(){
+    super();
+    this.state = {
+      email: "",
+      password: "",
+      confirmPassword: "",
+    }
+    this.onSignUp = this.onSignUp.bind(this);
+  }
+
+  onSignUp(navigation){
+    const expression = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+    if(this.state.password.length > 8 && expression.test(this.state.email.toLowerCase())){
+      if (this.state.password == this.state.confirmPassword){
+        this.props.onSignUp(this.state.email);
+        navigation.navigate('Details');
+      }
+      else{
+        alert("Different passwords");
+      }
+    }
+    else{
+      alert("Invalid email or password");
+    }
+  }
+
+  render(){
+    const { navigation } = this.props;
+    return (
+      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+        <Text style={{ fontSize: 30, height:"10%" }}>Sign Up</Text>
+        <View style={styles.inputView} >
+          <TextInput  
+            style={styles.inputText}
+            autoCorrect={false}
+            autoCapitalize = 'none'
+            placeholder="Email..." 
+            placeholderTextColor="#003f5c"
+            onChangeText={text => this.setState({email: text})}
+          />
+        </View>
+        <View style={styles.inputView} >
+          <TextInput
+              secureTextEntry={true}
+              autoCorrect={false}  
+              style={styles.inputText}
+              placeholder="Password..." 
+              placeholderTextColor="#003f5c"
+              onChangeText={text => this.setState({password:text})}
+          />
+        </View>
+        <View style={styles.inputView} >
+          <TextInput
+              secureTextEntry={true}
+              autoCorrect={false}  
+              style={styles.inputText}
+              placeholder="Confirm Password..." 
+              placeholderTextColor="#003f5c"
+              onChangeText={text => this.setState({confirmPassword:text})}
+          />
+        </View>
```

Comment:
Podes passar o style para o TextInput e assim evitamos ter mais elementos para fazer render
```suggestion
          <TextInput
              style={styles.inputView}
              secureTextEntry={true}
              autoCorrect={false}  
              style={styles.inputText}
              placeholder="Confirm Password..." 
              placeholderTextColor="#003f5c"
              onChangeText={text => this.setState({confirmPassword:text})}
          />
```